src/assets/TokenRatesController.ts

```diff
@@ -170,21 +214,43 @@ export class TokenRatesController extends BaseController<
    * @returns Promise resolving when this operation completes
    */
   async updateExchangeRates() {
-    if (this.tokenList.length === 0) {
+    if (this.tokenList.length === 0 || this.disabled) {
       return;
     }
-    const newContractExchangeRates: { [address: string]: number } = {};
-    const { nativeCurrency } = this.config;
-    const pairs = this.tokenList.map((token) => token.address).join(',');
-    const query = `contract_addresses=${pairs}&vs_currencies=${nativeCurrency.toLowerCase()}`;
-    const prices = await this.fetchExchangeRate(query);
-    this.tokenList.forEach((token) => {
-      const address = toChecksumAddress(token.address);
-      const price = prices[token.address.toLowerCase()];
-      newContractExchangeRates[address] = price
-        ? price[nativeCurrency.toLowerCase()]
-        : 0;
-    });
+    const { nativeCurrency, chainId } = this.config;
+    const { chainSlugIdentifier } = this.state;
+
+    let chainSlug;
+
+    if (!chainSlugIdentifier) {
+      try {
+        chainSlug = await this.getChainSlugIdentifier(chainId);
+        if (!chainSlug) {
+          this.update({ chainSlugIdentifier: null });
+        }
+      } catch {
+        this.update({ chainSlugIdentifier: undefined });
+      }
+    }
```

Comment:
Here's the main question. 

Since now there is an extra chainId parameter, we need to hit two endpoints
1. Get the chain slug given the chain id
2. Use the chain slug to recover the prices, like it was done before

**Should we "remember" the slug?** There are two three cases:
1. The request fails
2. The request is successful but the requested chain id is not listed on the api
3. The request is successful and we have a slug we can use

If we don't have the slug stored, this will always make two request when fetching prices.
If we store the slug, we must be able to define when to invalidate it in case the API changes, for example: it supports a new chain which we stored as unsupported

Reply:
Maybe we store the slug when we are able to get it and then wrap the `fetchExchangeRate` in a try/catch in case the slug becomes invalid and attempt to refetch it if that's the case?

Reply:
How would we manage the case the slug does not exist and later it does? Example: We would store it as `null`, so we don't even attempt to fetch the rates, but if the slug is supported in the future, how would we invalidate that `null`?

Reply:
Maybe I'm not understanding but if the chainSlugIdentifier is null we try will try to fetch it again as the code stands? Its not ideal because we are not able to cut out that extra network request by storing it as unsupported but I can't think of a way around that right now - unless they only update the api when they change the endpoint (v3 -> v4)?

Reply:
What we are doing on `swaps-controller` to hold values for a certain amount of time is to store pairs `(value, timestamp)` and compare `timestamp`, which represents the last time `value` was fetched, to a time threshold.

This way if `value` is `null` but `Date.now() - timestamp` is less than a threshold, we consider the network is not supported, otherwise we attempt to fetch the slug and update the pair `(value, timestamp = Date.now())`.

The same works when `value` is `a-network-slug` that might become unsupported, but we check only after we are over the time threshold.

Does it make sense?

Reply:
Yes that makes sense. So we're weighing the trade-offs of optimizing for fewer network requests and, if we choose to go that route, possibly not fetching exchange rates when they are available until the unsupported flag expires according to whatever time threshold we set. I don't have a strong intuition of which way to go with this. What do you think @Gudahtt?